services/ui-src/src/components/reports/DrawerReportPage.tsx

```diff
@@ -96,21 +98,39 @@ export const DrawerReportPage = ({ route }: Props) => {
     onClose();
   };
 
-  const entityRows = (entities: EntityShape[]) =>
-    entities.map((entity) => (
-      <Flex key={entity.id} sx={sx.entityRow}>
-        <Heading as="h4" sx={sx.entityName}>
-          {entity.name}
-        </Heading>
-        <Button
-          sx={sx.enterButton}
-          onClick={() => openRowDrawer(entity)}
-          variant="outline"
-        >
-          Enter
-        </Button>
-      </Flex>
-    ));
+  const entityRows = (entities: EntityShape[]) => {
+    return entities.map((entity) => {
+      /*
+       * If the entity has any data from any drawerForms fields,
+       * it must have been completed at some point. So simply check to
+       * see if the first drawerForm field has been entered and saved
+       * to the entity
+       */
+      const isEntityCompleted = drawerForm.fields?.[0]?.id in entity;
```

Comment:
Suggestion: We could probably future-proof this and make it so we don't need the comment if we just check for each of the fields in the entity with something like this:

```suggestion
      const isEntityCompleted = drawerForm.fields?.forEach((field: whatever) => field.id in entity);
```

That way we're explicitly checking for full completion. Just a thought.

Reply:
Paired it down to:

`const isEntityCompleted = drawerForm.fields?.every((field: FormField) => field.id in entity);`

I think that gets at what you were looking for